Cython/Compiler/Optimize.py

```diff
@@ -409,12 +409,12 @@ def make_length_call():
     PyBytes_AS_STRING_func_type = PyrexTypes.CFuncType(
         PyrexTypes.c_char_ptr_type, [
             PyrexTypes.CFuncTypeArg("s", Builtin.bytes_type, None)
-            ])
+            ], exception_value="NULL")
```

Comment:
I'd quite like some way to enable the exception check only when a macro is defined (to avoid generating the code on non-limited api cases). Probably an improvement for the future rather than this PR though.